schematic/manifest/generator.py

```diff
@@ -1254,18 +1256,19 @@ def get_empty_manifest(self, json_schema_filepath=None):
         )
         return manifest_url
 
-    def _get_missing_columns(self, wb_header, manifest_df):
-        # get headers from sheet 
-        manifest_df_header = manifest_df.columns
-
-        # find missing columns present in existing manifest but missing in latest schema
-        out_of_schema_columns = set(manifest_df_header) - set(wb_header)
-
-        return out_of_schema_columns
+    def _get_missing_columns(self, headers_1:list , headers_2:list) -> list:
+        """Compare two colunm sets and get cols that are in headers_1, but not headers_2
+        Args:
+            headers_1 (list): list of column headers
+            headers_2 (list): list of column headers
+        Returns: 
+            list: column headers in headers_1 but not headers_2
 
+        """
+        return set(headers_1) - set(headers_2)
 
     def set_dataframe_by_url(
-        self, manifest_url: str, manifest_df: pd.DataFrame
+        self, manifest_url: str, manifest_df: pd.DataFrame, out_of_schema_columns=None,
     ) -> ps.Spreadsheet:
```

Comment:
add type hinting of `out_of_schema_columns` and add a brief description of the parameter in the docstring.